sdk/storage/microsoft-azure-storage-blob/pom.xml

```diff
@@ -95,6 +95,13 @@
       <version>2.4-M1-groovy-4.0</version> <!-- {x-version-update;org.spockframework:spock-core;external_dependency} -->
       <scope>test</scope>
     </dependency>
+    <!--The Groovy version that Spock Core brings in (4.0.6) does not support Java 21, so we need to override it with a version that does.-->
+    <dependency>
```

Comment:
Same comment about being a spock-core upgrade

Reply:
Check my earlier answer about this.